shell/platform/android/test/io/flutter/embedding/android/FlutterActivityAndFragmentDelegateTest.java

```diff
@@ -1189,6 +1191,36 @@ public void usesFlutterEngineGroup() {
     assertEquals(engineUnderTest, mockFlutterEngine);
   }
 
+  @Test
+  public void pushesRouteInformationAfterFirstFrameIfApplicable() {
```

Comment:
Is it possible to add a test that ensures the route is not pushed the second time? Basically does hasPushedRouteInformation work.